insert-interval/wogha95.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * TC: O(N)
+ * 1번에서 N만큼 순회
+ * 2번에서 최대 N만큼 순회
+ *
+ * SC: O(1)
+ * 겹치는 interval의 시작, 끝 index를 저장하는 변수만 사용
+ *
+ * N: intervals.length
+ */
+
+/**
+ * @param {number[][]} intervals
+ * @param {number[]} newInterval
+ * @return {number[][]}
+ */
+var insert = function (intervals, newInterval) {
+  if (intervals.length === 0) {
+    return [newInterval];
+  }
+  if (newInterval[1] < intervals[0][0]) {
+    return [newInterval, ...intervals];
+  }
+  if (intervals[intervals.length - 1][1] < newInterval[0]) {
+    return [...intervals, newInterval];
+  }
+
+  // 1. 겹치는 interval의 시작, 끝 index를 구한다.
+  let overLappingStartIndex = intervals.length - 1;
+  let overLappingEndIndex = 0;
+  for (let index = 0; index < intervals.length; index++) {
+    const interval = intervals[index];
+    if (
+      (interval[0] <= newInterval[0] && newInterval[0] <= interval[1]) ||
+      newInterval[0] < interval[0]
+    ) {
+      overLappingStartIndex = Math.min(overLappingStartIndex, index);
+    }
+    if (
+      (interval[0] <= newInterval[1] && newInterval[1] <= interval[1]) ||
+      interval[1] < newInterval[1]
+    ) {
+      overLappingEndIndex = Math.max(overLappingEndIndex, index);
+    }
```

Comment:
intervals가 이미 정렬되어 있다는 사실을 고려하면, if 조건을 훨씬 간소화 시킬 수 있지 않을까요?

```suggestion
    if (newInterval[0] <= interval[1]) {
      overLappingStartIndex = Math.min(overLappingStartIndex, index);
    }
    if (interval[0] <= newInterval[1]) {
      overLappingEndIndex = Math.max(overLappingEndIndex, index);
    }
```

Reply:
각 범위의 양끝을 각각 생각하느라 통합된 범위로 보질 못했습니다..
말씀주신대로 조건문처리하는 것이 더 적절하다고 생각됩니다!!
감사합니다~!